pkg/azure/api/providerspec.go

```diff
@@ -109,6 +113,15 @@ type AzureVirtualMachineProperties struct {
 	DiagnosticsProfile *AzureDiagnosticsProfile `json:"diagnosticsProfile,omitempty"`
 	// Deprecated: Use either AvailabilitySet or VirtualMachineScaleSet instead
 	MachineSet *AzureMachineSetConfig `json:"machineSet,omitempty"`
+	// SecurityProfile specifies the security profile to be used for the virtual machine.
+	SecurityProfile *AzureSecurityProfile `json:"securityProfile,omitempty"`
+}
+
+// AzureSecurityProfile specifies the security profile to be used for the virtual machine.
+type AzureSecurityProfile struct {
+	// SecurityType specifies the SecurityType attribute of the virtual machine.
+	// Only "ConfidentialVM" is supported for now.
+	SecurityType string `json:"securityType,omitempty"`
```

Comment:
In the validation you have the following check:
```go
if prof.SecurityType != string(armcompute.SecurityTypesConfidentialVM) {
		allErrs = append(allErrs, field.Invalid(fldPath.Child("securityType"), prof.SecurityType, "security type not supported"))
```
This check effectively makes this field as mandatory if the `AzureSecutityProfile` is set. Just wondering if `omitempty` is in line with this? In addition, there seems to be a convention being followed - for `non-mandatory` fields a pointer is used. This is not followed here.

Reply:
> This check effectively makes this field as mandatory if the AzureSecutityProfile is set.

Correct. We only looked into supporting `ConfidentialVMs` with this PR. Consequently, we only know how to support this one SecurityType and we validate against that [1].

> Just wondering if omitempty is in line with this?

The `omitempty` will make it so that it doesn't appear in the json, but will still get the "zero" value of a string. If you want an example the `Namespace` in k8s objects is defined:

```
Namespace string `json:"namespace,omitempty" protobuf:"bytes,3,opt,name=namespace"`
````
So it's not unheard of. But we can update this 👍  

> In addition, there seems to be a convention being followed - for non-mandatory fields a pointer is used. This is not followed here.

```
SecurityProfile *AzureSecurityProfile `json:"securityProfile,omitempty"`
```
Security profile struct is declared with a pointer which makes it optional overall. If you do declare it in your spec though, you need to have a valid securityType used. 

[1]++ The "mandatory" part comes as a consequence of only support that one "securityType", which effectively makes it mandatory is you plan to specify a SecurityProfile.
```
SecurityType string
```
can always be lax'ed to a pointer in the future.